api/internal/core/entity/interface.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package entity
+
+type GetPlugins interface {
+	GetPlugins() map[string]interface{}
+}
+
+type GetBaseInfo interface {
+	GetBaseInfo() *BaseInfo
+}
+
+func (r *Route) GetPlugins() map[string]interface{} {
+	return r.Plugins
+}
+
+func (s *Service) GetPlugins() map[string]interface{} {
+	return s.Plugins
+}
+
+func (c *Consumer) GetPlugins() map[string]interface{} {
+	return c.Plugins
+}
+
+func (g *GlobalPlugins) GetPlugins() map[string]interface{} {
+	return g.Plugins
+}
+
+func (p *PluginConfig) GetPlugins() map[string]interface{} {
+	return p.Plugins
+}
+
+func (r *Route) GetBaseInfo() *BaseInfo {
+	return &r.BaseInfo
+}
+
+func (u *Upstream) GetBaseInfo() *BaseInfo {
+	return &u.BaseInfo
+}
+
+func (s *Service) GetBaseInfo() *BaseInfo {
+	return &s.BaseInfo
+}
+
+func (c *Consumer) GetBaseInfo() *BaseInfo {
+	return &BaseInfo{
+		ID:         c.Username,
+		CreateTime: c.CreateTime,
+		UpdateTime: c.UpdateTime,
+	}
+}
+
+func (s *SSL) GetBaseInfo() *BaseInfo {
+	return &s.BaseInfo
+}
+
+func (g *GlobalPlugins) GetBaseInfo() *BaseInfo {
+	return &g.BaseInfo
+}
+
+func (p *PluginConfig) GetBaseInfo() *BaseInfo {
+	return &p.BaseInfo
+}
+
+func (s *ServerInfo) GetBaseInfo() *BaseInfo {
+	return &s.BaseInfo
+}
+
+func (p *Proto) GetBaseInfo() *BaseInfo {
```

Comment:
What are the benefits of defining a method `GetBaseInfo` for each struct?

Reply:
It just replaces the original `BaseInfoGetter` and `BaseInfoSetter`. Because they are all operations on the structure pointer, there is no need to implement them twice. Now use a single method to cover all the data structures, making it easier to get `BaseInfo`.

Reply:
I think the risk is greater than the benefit. Because when adding a new struct, may forget to add this method for it, resulting in an error when processing the request.

Reply:
Yes, I can understand.
But I found some problems with the previous code. Just like this,
[https://github.com/apache/apisix-dashboard/blob/a3928d8bbcbaec089458db01eb7a380f20fb33ae/api/internal/core/store/store.go#L174-L175](https://github.com/apache/apisix-dashboard/blob/a3928d8bbcbaec089458db01eb7a380f20fb33ae/api/internal/core/store/store.go#L174-L175)
Although type conversion is performed here, none of the data entities actually implement the getter interface. When executing list query, this will cause panic in the program, so I am very confused.

Reply:
`entity.BaseInfo` is the implement

Reply:
Oh, I get it. I will modify it this way.

Reply:
@nic-chen I have removed the implementation of `GetBaseInfo` in the data entity, and now only the implementation of `GetPlugins` is left.